### PR TITLE
add waitForBucketReadAccess

### PR DESCRIPTION
**Pull request by dvoet**
**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

After merging, _even if you haven't bumped the version_: 
- [ ] Update `README.md` and the `CHANGELOG.md` for any libs you changed with the new dependency string. The git hash is the same short version you see in GitHub (i.e. seven characters). You can commit these changes straight to develop.

### CODE REVIEW

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Orchestration.scala [hunk 1]
```diff
@@ -139,6 +139,17 @@ trait Orchestration extends RestClient with LazyLogging with SprayJsonSupport wi
       logger.info(s"Setting attributes for workspace: $namespace/$name $attributes")
       patchRequest(apiUrl(s"api/workspaces/$namespace/$name/setAttributes"), attributes)
     }
+
+    def waitForBucketReadAccess(namespace: String, name: String)(implicit token: AuthToken): Unit = {
+      logger.info(s"checking bucket read access for workspace: $namespace/$name")
+      Retry.retry(10.seconds, 10.minutes)({
```
Comment (breilly2) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Orchestration.scala]: How long does this typically take?

**Review by breilly2 — APPROVED** [review]
👍

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Orchestration.scala [hunk 2]
```diff
@@ -139,6 +139,17 @@ trait Orchestration extends RestClient with LazyLogging with SprayJsonSupport wi
       logger.info(s"Setting attributes for workspace: $namespace/$name $attributes")
       patchRequest(apiUrl(s"api/workspaces/$namespace/$name/setAttributes"), attributes)
     }
+
+    def waitForBucketReadAccess(namespace: String, name: String)(implicit token: AuthToken): Unit = {
```
Comment (breilly2) [inline serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Orchestration.scala]: Actually, can you add some docs with guidance on when this is necessary? Looking at the tests, most are launching an analysis but not all so I'm not sure I'd know when to use it.

File: serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Orchestration.scala (hunk 1 above)
Comment (dvoet) [reply serviceTest/src/test/scala/org/broadinstitute/dsde/workbench/service/Orchestration.scala]: I have only seen this be a problem for the first pet service account of a new project. In that case I have seen it take 5 minutes.

**Pull request merged**